packages/telemetry/test/flight-recorder.test.js

```diff
@@ -17,12 +18,20 @@ const bufferTests = test.macro(
   async (t, input) => {
     const BUFFER_SIZE = 512;
 
-    const { name: tmpFile, removeCallback } = tmp.fileSync();
+    const {
+      name: tmpFile,
+      fd,
+      removeCallback,
+    } = tmp.fileSync({ detachDescriptor: true });
+    t.teardown(removeCallback);
+    const fileHandle = /** @type {import('fs/promises').FileHandle} */ ({
+      close: promisify(fs.close.bind(fs, fd)),
+    });
     const { readCircBuf, writeCircBuf } = await input.makeBuffer({
       circularBufferSize: BUFFER_SIZE,
       circularBufferFilename: tmpFile,
     });
-    const slogSender = makeSlogSenderFromBuffer({ writeCircBuf });
+    const slogSender = makeSlogSenderFromBuffer({ fileHandle, writeCircBuf });
```

Comment:
should we add a shutdown?

```suggestion
    const slogSender = makeSlogSenderFromBuffer({ fileHandle, writeCircBuf });
    t.teardown(slogsender.shutdown);
```

Reply:
Yeah, I like that.